core/src/com/unciv/ui/screens/victoryscreen/VictoryScreen.kt

```diff
@@ -187,6 +197,43 @@ class VictoryScreen(val worldScreen: WorldScreen) : PickerScreen() {
         else contentsTable.add(buildRankingsTable(majorCivs))
     }
 
+    private fun setReplayTable() {
+        val replayTable = Table().apply { defaults().pad(10f) }
+        val yearLabel = "".toLabel()
+        replayTable.add(yearLabel).row()
+        val replayMap = ReplayMap(gameInfo.tileMap)
+        replayTable.add(replayMap).row()
+
+        var nextTurn = 0
+        val finalTurn = UncivGame.Current.gameInfo?.turns ?: 0
+        Timer.schedule(
+            object : Timer.Task() {
+                override fun run() {
+                    updateReplayTable(yearLabel, replayMap, nextTurn++)
+                }
+            }, 0.0f,
+            // A game of 600 rounds will take one minute.
+            0.1f,
+            // End at the last turn.
+            finalTurn
+        )
```

Comment:
The timer does not stop when you change tabs or close the victory screen

Reply:
I added code to cancel the Timer.Task if the Replay is opened again and also when the VictoryScreen is closed. I didn't find an elegant solution though to do it when the tab is switched. I tried to add onClick for the whole tab bar, but unfortunately that gets executed after the ReplayMap is initialized unfortunately, so the task will be canceled immediately. Any idea how to do it nicer than add it to every other onClick handler? I think it might be acceptable to have it live on in the scope of the VictoryScreen once since it is also finite. WDYT?

Reply:
You can call `replayTimer?.cancel()` in each `setSomeTable()` fun, where `contentsTable.clear()` is called

Reply:
Done